conjure/ui/views/steps.py

```diff
@@ -1,60 +1,84 @@
 from conjure.ui.widgets.step import StepWidget
-from ubuntui.utils import Padding
+from ubuntui.utils import Padding, Color
 from ubuntui.widgets.table import Table
+from ubuntui.widgets.buttons import submit_btn, done_btn
 from urwid import WidgetWrap
 import random
 
 
 class StepsView(WidgetWrap):
 
-    def __init__(self, app, steps):
+    def __init__(self, app, steps, cb=None):
+        """ init
+
+        Arguments:
+        cb: process step callback
+        """
         self.app = app
         self.table = Table()
+        self.cb = cb
 
         self.steps_queue = steps
         for k, v in self.steps_queue.items():
-            self.steps_queue[k] = StepWidget(v)
+            meta = StepWidget(v['step_metadata'])
```

Comment:
calling the StepWidget 'meta' is a pretty generic name, makes it hard to follow the rest of the code.